### ISSUE TITLE
Assertion `time >= frame.time()' failed in read_raw_data_for_frame, during replay

### ISSUE THREAD

**Issue by mbrubeck** [mozilla/rr#1639]
I got this assertion when running "rr replay" on one recording of a 64-bit Servo process:

`rr: /home/mbrubeck/src/rr/rr/src/TraceStream.cc:456: bool TraceReader::read_raw_data_for_frame(const TraceFrame&, TraceReader::RawData&): Assertion `time >= frame.time()' failed`

**Comment by mbrubeck** [comment]
This was using rr revision 0e3a9c9c72a65cd30aac13b65d60d0b65f135e96

**Comment by rocallahan** [comment]
Something probably killed rr prematurely with kill -9 or whatever. Make sure rr terminates normally or dies with a SIGINT or SIGTERM.

**Comment by bgirard** [comment]
Sounds like something that is likely to happen a lot in practice for whatever reason. Perhaps this can use a more user friendly message to let them know the trace is incomplete?

**Comment by rocallahan** [comment]
Definitely, though I also need to think about making the problem go away altogether. Moving parts of trace writing to another process could probably fix it.
